features/step_definitions/shf_documents_steps.rb

```diff
@@ -0,0 +1,20 @@
+When(/^I choose a shf-document named "([^"]*)" to upload$/) do | filename |
+  page.attach_file "shf_document[actual_file]", File.join(Rails.root, 'spec', 'fixtures','uploaded_files', filename)
+end
+
+
+And(/^I should see (\d+) shf-documents listed$/) do |number|
+  expect(page).to have_selector('.shf-document', count: number)
```

Comment:
that is _really_ good info.  I knew there were issues with methods not actually waiting for items to show up.  Glad to know the way to make sure you wait for them is actually this easy. :-)